File: Nx3Java/java/src/nx3/PComplexDistancesCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PComplexDistancesCalculator extends haxe.lang.HxObject
{
	public    PComplexDistancesCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PComplexDistancesCalculator()
	{
		nx3.PComplexDistancesCalculator.__hx_ctor_nx3_PComplexDistancesCalculator(this);
	}
	
	
	public static   void __hx_ctor_nx3_PComplexDistancesCalculator(nx3.PComplexDistancesCalculator __temp_me112)
	{
		{
		}
		
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PComplexDistancesCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PComplexDistancesCalculator();
	}
	
	
	public   double getDistance(nx3.PComplex leftComplex, nx3.PComplex rightComplex)
	{
		haxe.root.Array<nx3.geom.Rectangle> leftBaseRects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{leftComplex.getBaseRect()});
		haxe.root.Array<nx3.geom.Rectangle> rightBaseRects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{rightComplex.getBaseRect()});
		double minDistance = nx3.geom.RectanglesTools.getXIntersection(leftBaseRects, rightBaseRects);
		haxe.root.Array<nx3.geom.Rectangle> leftRects = leftComplex.getAllRects();
		haxe.root.Array<nx3.geom.Rectangle> rightRects = rightComplex.getAllRects();
		double objDistance = nx3.geom.RectanglesTools.getXIntersection(leftRects, rightRects);
		double objDistanceMargin = ( objDistance + 0.6 );
		return java.lang.Math.max(minDistance, objDistanceMargin);
	}
	
	
	public   haxe.root.Array<nx3.geom.Rectangle> getRects(nx3.PComplex complex)
	{
		haxe.root.Array<nx3.geom.Rectangle> rects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{});
		rects.concat(complex.getHeadsRects());
		rects.concat(complex.getStavesRects());
		rects.concat(complex.getSignsRects());
		return rects;
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef685 = true;
			switch (field.hashCode())
			{
				case 1964214617:
				{
					if (field.equals("getRects")) 
					{
						__temp_executeDef685 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getRects"))) );
					}
					
					break;
				}
				
				
				case -1928607381:
				{
					if (field.equals("getDistance")) 
					{
						__temp_executeDef685 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getDistance"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef685) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef686 = true;
			switch (field.hashCode())
			{
				case 1964214617:
				{
					if (field.equals("getRects")) 
					{
						__temp_executeDef686 = false;
						return this.getRects(((nx3.PComplex) (dynargs.__get(0)) ));
					}
					
					break;
				}
				
				
				case -1928607381:
				{
					if (field.equals("getDistance")) 
					{
						__temp_executeDef686 = false;
						return this.getDistance(((nx3.PComplex) (dynargs.__get(0)) ), ((nx3.PComplex) (dynargs.__get(1)) ));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef686) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
}

File: Nx3CS/bin/test/src/nx3/PComplexDistancesCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PComplexDistancesCalculator : global::haxe.lang.HxObject {
		public    PComplexDistancesCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PComplexDistancesCalculator(){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				global::nx3.PComplexDistancesCalculator.__hx_ctor_nx3_PComplexDistancesCalculator(this);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PComplexDistancesCalculator(global::nx3.PComplexDistancesCalculator __temp_me97){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				return new global::nx3.PComplexDistancesCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				return new global::nx3.PComplexDistancesCalculator();
			}
			#line default
		}
		
		
		public virtual   double getDistance(global::nx3.PComplex leftComplex, global::nx3.PComplex rightComplex){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				global::Array<object> leftBaseRects = new global::Array<object>(new object[]{leftComplex.getBaseRect()});
				global::Array<object> rightBaseRects = new global::Array<object>(new object[]{rightComplex.getBaseRect()});
				double minDistance = global::nx3.geom.RectanglesTools.getXIntersection(leftBaseRects, rightBaseRects);
				#line 25 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				global::Array<object> leftRects = leftComplex.getAllRects();
				global::Array<object> rightRects = rightComplex.getAllRects();
				double objDistance = global::nx3.geom.RectanglesTools.getXIntersection(leftRects, rightRects);
				#line 29 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				double objDistanceMargin = ( objDistance + 0.6 );
				#line 31 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				return global::System.Math.Max(((double) (minDistance) ), ((double) (objDistanceMargin) ));
			}
			#line default
		}
		
		
		public virtual   global::Array<object> getRects(global::nx3.PComplex complex){
			unchecked {
				#line 37 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				global::Array<object> rects = new global::Array<object>(new object[]{});
				#line 39 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				rects.concat(complex.getHeadsRects());
				rects.concat(complex.getStavesRects());
				rects.concat(complex.getSignsRects());
				#line 43 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				return rects;
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				switch (hash){
					case 1184055897:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getRects") ), ((int) (1184055897) ))) );
					}
					
					
					case 300766699:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getDistance") ), ((int) (300766699) ))) );
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
				switch (hash){
					case 1184055897:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return this.getRects(((global::nx3.PComplex) (dynargs[0]) ));
					}
					
					
					case 300766699:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return this.getDistance(((global::nx3.PComplex) (dynargs[0]) ), ((global::nx3.PComplex) (dynargs[1]) ));
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PComplexDistancesCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
	}
}


